lib/pagy/extras/infinite.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+require 'pagy/extras/shared'
+
+class Pagy
+  module Backend ; private         # the whole module is private so no problem with including it in a controller
+
+    # Return Pagy object and items
+    def pagy_infinite(collection, vars={})
+      pagy = Pagy.new(pagy_infinite_get_vars(collection, vars))
+      return pagy, pagy_get_items(collection, pagy)
+    end
+
+    # Sub-method called only by #pagy: here for easy customization of variables by overriding
+    def  pagy_infinite_get_vars(collection, vars)
+      current_page = vars[:page] || (params[vars[:page_param] || Pagy::VARS[:page_param]] || 1).to_i
+      per_page_items = vars[:items] || Pagy::VARS[:items]
+      infinite_count = current_page * per_page_items
+      if collection.limit(per_page_items).offset(infinite_count).exists?
+        infinite_count += 1
+      end
+
```

Comment:
Since the extra is not dedicated to any particular collection type, you should not assume that `limit` and `offset` will be always used for all the collections that might benefit from the extra. You should override the `pagy_get_items` or extracting the collection handling in its own method.

Reply:
I need to think this over. Don't quite understand relation with `pagy_get_items`

Reply:
Indeed my comment was misleading. I didn't understood your code. :) Forget about it.

So... what you need is checking if there is a full page of items (+1) or not, and that could be done avoiding to use `limit` and `offset` in the method by focusing on the other known variables, which would make the method collection-agnostic.

For example: if you select a page of `:items + 1` and you check that you retrieved `:items + 1` items, then you can pop the last item, so the item number will be correct. And if you pop the last item that means that you can set the `infinite_count += 1`. Instead if the retrieved items is less than `:items + 1` you don't need to pop the last item nor set the `infinite_count`.

Then you should override the `pagy_get_items` and reorganize the `pagy_infinite` accordingly... but I leave it to you for now. ;)